dbms/src/Storages/Page/V3/BlobStore.cpp

```diff
@@ -119,7 +114,14 @@ void BlobStore::registerPaths()
 
 void BlobStore::reloadConfig(const BlobStore::Config & rhs)
 {
-    config.file_limit_size = rhs.file_limit_size;
+    /// Don't reload config.file_limit_size here. Because there may be a potential race between write thread and gc thread.
+    /// For write thread, it will ignore all BlobFile which capacity is larger than config.file_limit_size.
+    /// For gc thread, if the contents of a BlobFile with capacity larger than config.file_limit_size is all removed, it will remove the BlobFile.
+    /// So if you reload config.file_limit_size with a larger value,
+    /// the write thread may see the change and select a BlobFile which capacity is larger than old config.file_limit_size but smaller than new value for write,
+    /// but the gc thread may not see the change and still think the BlobFile's capacity is larger than config.file_limit_size and remove it if all its contents have been removed.
```

Comment:
I thought we already guarantee that a blob file selected as GC candidate won't be chosen as a writable file?

Reply:
Actually it is not before this commit: https://github.com/pingcap/tiflash/pull/5544/commits/af71d61b2159aef547809e3e9d0cbdcca7fc5be5, but I figured it out according to @JaySon-Huang' s previous comment. So the comments is deleted now.